blockchain/api.go

```diff
@@ -1364,24 +1369,43 @@ func (api *BasicEventsAPI) GetLastBlock(ctx context.Context) (uint64, error) {
 
 func (api *BasicEventsAPI) GetEvents(ctx context.Context, filter *EventFilter) (chan *Event, error) {
 	var out = make(chan *Event, 128)
+	// Number of blocks already present in blockchain that we will read in batches.
+	lastBlock, err := api.GetLastBlock(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest block number: %v", err)
+	}
+
+	if filter.FromBlock.Uint64() > lastBlock {
+		return nil, errors.New("FromBlock value is greater than current number of blocks")
+	}
+
 	go func() {
 		var (
-			lastLogBlockNumber = filter.fromBlock.Uint64()
-			fromBlock          = filter.fromBlock.Uint64()
-			tk                 = time.NewTicker(time.Second)
+			blocksLeft         = int64(lastBlock) - filter.FromBlock.Int64()
+			lastLogBlockNumber = filter.FromBlock.Uint64()
+			fromBlock          = filter.FromBlock.Uint64()
+			toBlock            = big.NewInt(0)
+			tk                 = time.NewTicker(time.Millisecond * 100)
 		)
 		for {
 			select {
 			case <-ctx.Done():
 				close(out)
 				return
 			case <-tk.C:
+				if blocksLeft > 0 {
```

Comment:
rather complex logic, why not just update toBlock constantly? does handle return error if toBlock is larger than known block?